File: Assignments/Weeklies/EnfeiZhang_GrittyP3/Question2/Sphere.java
/**Enfei Zhang
  March 18, 2018
  Ms. Krasteva
  This is a subclass of the abstract class solids.
 */


public class Sphere extends Solid{

    private double radius;
    private final double pi = Math.PI;

    public Sphere (String n, double r){
        super (n);
        radius = r;
    }

    @Override
    public double volume (){
        return (4/3)*pi*Math.pow(radius, 3);
    }

    @Override
    public double surfaceArea(){
        return 4*pi*Math.pow(radius, 2);
    }

    @Override
    public double perimeter(){
        return 2*pi*radius;
    }
}

File: Assignments/Weeklies/EnfeiZhang_GrittyP3/Question2/Sphere.kt
/*Enfei Zhang
  March 18, 2018
  Ms. Krasteva
  This is a subclass of the abstract class solids.
 */

class Sphere(n: String, private val radius: Double) : Solid(n) {
    private val pi = Math.PI

    override fun volume(): Double {
        return (4 / 3).toDouble() * pi * Math.pow(radius, 3.0)
    }

    override fun surfaceArea(): Double {
        return 4.0 * pi * Math.pow(radius, 2.0)
    }

    override fun perimeter(): Double {
        return 2.0 * pi * radius
    }
}
